autoscale-source-kubernetes/src/main/java/com/hpe/caf/autoscale/source/kubernetes/K8sServiceSource.java

```diff
@@ -0,0 +1,135 @@
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hpe.caf.autoscale.source.kubernetes;
+
+import com.hpe.caf.api.HealthResult;
+import com.hpe.caf.api.HealthStatus;
+import com.hpe.caf.api.autoscale.ScalerException;
+import com.hpe.caf.api.autoscale.ScalingConfiguration;
+import com.hpe.caf.api.autoscale.ServiceSource;
+import com.hpe.caf.autoscale.K8sAutoscaleConfiguration;
+import io.kubernetes.client.extended.kubectl.Kubectl;
+import io.kubernetes.client.extended.kubectl.exception.KubectlException;
+import io.kubernetes.client.openapi.models.V1Deployment;
+import io.kubernetes.client.openapi.models.V1ObjectMeta;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class K8sServiceSource implements ServiceSource
+{
+    private static final Logger LOG = LoggerFactory.getLogger(K8sServiceSource.class);
+    
+    private final K8sAutoscaleConfiguration config;
+
+    public K8sServiceSource(final K8sAutoscaleConfiguration config)
+    {
+        this.config = Objects.requireNonNull(config);
+    }
+
+    @Override
+    public Set<ScalingConfiguration> getServices()
+        throws ScalerException
+    {
+        try {
+            return getScalingConfiguration();
+        } catch (NumberFormatException e) {
+            throw new ScalerException("Error parsing Deployment label", e);
+        } catch (KubectlException e) {
+            throw new ScalerException("Error loading deployments", e);
+        }
+    }
+
+    private Set<ScalingConfiguration> getScalingConfiguration() throws KubectlException
+    {
+        final Set<ScalingConfiguration> scalingConfigurations = new HashSet<>();
+        for (final String namespace: config.getNamespacesArray()) {
+            scalingConfigurations.addAll(Kubectl.get(V1Deployment.class)
+                .namespace(namespace)
+                .execute()
+                .stream()
+                .filter(d -> isLabelledForScaling(d))
+                .map(d -> mapToScalingConfig(d, namespace))
+                .collect(Collectors.toSet()));
+        }
+        return scalingConfigurations;
+    }
+
+    private ScalingConfiguration mapToScalingConfig(final V1Deployment deployment, final String namespace)
+    {
+        final ScalingConfiguration cfg = new ScalingConfiguration();
+        final V1ObjectMeta metadata = deployment.getMetadata();
+        final Map<String, String> labels = metadata.getLabels();
```

Comment:
`.getMetadata()` is explicitly declared that it might return `null`.

IDE reporting warning because `metadata` is being deferenced without checking if it is `null`.

![image](https://user-images.githubusercontent.com/20627297/123203206-e4eb7900-d4ad-11eb-9fc2-03c14c35afe3.png)

Reply:
Done